packages/tree/src/utils/merkle-tree-sol.ts

```diff
@@ -259,9 +259,9 @@ export function splitToSubTrees<T extends Fp | BN>(
 ): T[][] {
   const subTreeSize = 1 << subTreeDepth
   const numOfSubTrees = Math.ceil(leaves.length / subTreeSize)
-  const subTrees: T[][] = Array(numOfSubTrees).fill(
-    Array(subTreeSize).fill(hasher.preHash[0]),
-  )
+  const subTrees: T[][] = Array(numOfSubTrees)
```

Comment:
I can't figure out what is different between these two code snippets. I've run both and they seem to output the same thing.

Why does this change fix the invalid root problem?

Reply:
Just tested against a testnet chain that had the invalid utxo root problem and this fixes solves the problem. Still not sure how.

Reply:
So I researched this more and `Array.prototype.fill` uses the same reference in each index when an object is passed. As a result using the `map` function with new arrays solves the problem.

This can be written as

```
const subTrees: T[][] = Array(numOfSubTrees)
.fill() // you can fill with undefined to make it more clear that this is just to make it iterable
.map(() => Array(subTreeSize).fill(hasher.preHash[0]))
```

Good find, just needs linting and then we can merge.

Reply:
> Why does this change fix the invalid root problem?

It is only problem with more than one `subTrees` created.

> So I researched this more and Array.prototype.fill uses the same reference in each index when an object is passed. As a result using the map function with new arrays solves the problem.

I also learn that this time! I feel bad that you could save your time if I was left what i found for this.

Updated code follow your suggestion. Thanks! 83b14d3